File: 프로그래머스/lv1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.kt
class Solution {
    fun solution(lottos: IntArray, win_nums: IntArray): IntArray {
        val trueCount = lottos.count { win_nums.contains(it) }
        val zeroCount = lottos.count { it == 0 }

        return intArrayOf(
            rankingLotto(trueCount + zeroCount),
            rankingLotto(trueCount)
        )
    }

    fun rankingLotto(number: Int): Int {
        return when(number) {
            6 -> 1
            5 -> 2
            4 -> 3
            3 -> 4
            2 -> 5
            else -> 6
        }
    }
}

File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.swift
func solution(_ lottos:[Int], _ win_nums:[Int]) -> [Int] {
    var answer = 0
    var zero = 0
    
    for lotto in lottos {
        if win_nums.contains(lotto) {
            answer -= 1
        } else if lotto == 0 {
            zero += 1
        }
    }
    
    if answer == 0 && zero == 0 {
        return [6-zero, 6]
    } else {
        return (answer == 0) ? [7-zero, 6] : [answer+7-zero, answer+7]
    }
}